### DISCUSSION TITLE
What version is app.rallly.co ?

### DISCUSSION THREAD

**Discussion by benjaoming** [lukevella/rallly#970] [Ideas]
Hi :wave: 

Thanks for the amazing project! And congratulation on a number of great UI decisions :1st_place_medal: 

I was wondering about what version app.rallly.co is running?

Maybe the version string can optionally be made visible in the footer?

(my wondering started when I read in the 3.0 release notes that "As of version 3, users must be logged in to create and manage polls", and then found that I can still create guest polls on the Demo site)

There can be differing views on this... but in a self-hosted setting, it can be a bad idea generally to state the exact version number. There might be a future security issue that's identified and fixed, and then it will be easy to identify self-hosted instances that are vulnerable.

**Comment by lukevella** [comment]
Hi @benjaoming, sorry for the late reply. The managed service isn't running a specific version number but rather it's synced up to the main branch of this repo. For self-hosting, a release is made intermittently when there are significant changes or important patches.

> (my wondering started when I read in the 3.0 release notes that "As of version 3, users must be logged in to create and manage polls", and then found that I can still create guest polls on the Demo site)

The self-hosted image is built with a feature flag that alters the final output of the app so that it is more suited for self-hosted use cases. For example, billing is disabled and all Pro features are available for free. Limiting poll creation to registered users for self-hosted instance was done for a number of reasons that aimed to make both my own and self-hosters' lives easier but its been made clear that many users are not happy about this.

> There can be differing views on this... but in a self-hosted setting, it can be a bad idea generally to state the exact version number. There might be a future security issue that's identified and fixed, and then it will be easy to identify self-hosted instances that are vulnerable.

That's a fair point. I'm open to suggestions on ways to display the version number without exposing self-hosted instance to security risks.

**Reply by benjaoming** [reply]
This sounds great! Can you reference the feature flag in the code?

Then I think we'd like to start using it, but I was wondering how to contribute back if starting to use it? Would you be interested in exposing it as a separate environment variable for the Docker setup? Or would you like if we start writing more documentation for using the self-hosted image, including mention of the feature flag?

**Reply by lukevella** [reply]
It's not intended for public consumption. Currently there is a [single build argument](https://github.com/lukevella/rallly/blob/main/.github/workflows/docker-image-version-release.yml#L42) that is used to switch the output of the build for the self-hosted image. If you don't use this build argument, you will end up with a version with only the free features enabled and a non-functional billing page.

Perhaps in the future I can offer more granular configuration of the instance but at the moment I would prefer to keep configuration as simple as possible to avoid taking on too much work in maintenance.

**Reply by benjaoming** [reply]
Sorry about leaving the discussion here open! Thanks for getting back! I take it as a polite "no thanks" to the contribution suggestion :wink: 

To the point of the main question, I think your answer was:

> The managed service isn't running a specific version number but rather it's synced up to the main branch of this repo

That's nice to know for me - now I can quickly test out features on the demo to get inspired.

> Limiting poll creation to registered users for self-hosted instance was done for a number of reasons that aimed to make both my own and self-hosters' lives easier but its been made clear that many users are not happy about this.

_**If**_ you're spending a lot of time trying to figure out compromises or somehow servicing self-hosting use-cases that you spend time understanding, then it might ultimately be easier to leave a bunch of things open to configuration through a public interface. This way, you might spend less time on communication by simply referring to documentation, and you can have more real-life production hardening done based on self-hosters' endless permutations of settings. This cannot be done in a CI.. and might ultimately turn out to be a benefit for the project for all users. `</hypthesis>`
